src/components/Header/Header.tsx

```diff
@@ -35,6 +40,32 @@ export const Header = chakra(() => {
   });
 
   const [loginOpen, setLoginOpen] = useState<boolean>(false);
+  const [patronDataReceived, setPatronDataReceived] = useState<boolean>(false);
+  const [patronName, setPatronName] = useState<string>("");
+
+  // Allows user to use esc key to close the login menu.
+  useEffect(() => {
+    const close = (e) => {
+      if (e.keyCode === 27) {
+        setLoginOpen(false);
```

Comment:
This is nice. We can reuse this for all the other button/menus.

Reply:
Adjusted to make work for more browsers.